### PR TITLE
Add support for Sitemap plugin, fix #6

### LINKED ISSUE #6: Add support for Sitemap plugin
This plugin should register a custom item types to the Page plugin to be able to add to the Sitemap plugin at least all channels urls and all topics urls.

Currently toying with it to build a tiny forum, I'll probably make a PR soon, if any suggestion on what should be or not.

### PR DESCRIPTION
**Pull request by RomainMazB**
This PR will add support for Sitemap plugin, it will register three items:
- All channels page
- Specific channel page
- All topics page

At first, I wanted to add specific topic page, but I'm not quite sure that this would be useful...

Need to be added:
- [x] Hide not public channels URL 
- [x] Hide not public topics URL
- [x] Translations keys
- [x] Suggestions?

### CODE REVIEW

File: Plugin.php [hunk 1]
```diff
@@ -91,6 +93,31 @@ public function boot()
...
+                return Topic::getMenuTypeInfo($type);
+            }
+        });
+
+        Event::listen('pages.menuitem.resolveItem', function($type, $item, $url, $theme) {
+            if ($type == 'forum-channel' || $type == 'all-forum-channels') {
+                return Channel::resolveMenuItem($item, $url, $theme);
+            } elseif ($type == 'all-forum-topics') {
+                return Topic::resolveMenuItem($item, $url, $theme);
+            }
```
Comment (LukeTowers) [inline Plugin.php:119]: Switch statements might be better here, @bennothommo any thoughts?

File: models/Channel.php [hunk 2]
```diff
@@ -78,8 +81,219 @@ class Channel extends Model
...
+     * - nesting - Boolean value indicating whether the item type supports nested items. Optional,
+     *   false if omitted.
+     * - dynamicItems - Boolean value indicating whether the item type could generate new menu items.
+     *   Optional, false if omitted.
+     * - cmsPages - a list of CMS pages (objects of the Cms\Classes\Page class), if the item type requires a CMS page reference to
+     *   resolve the item URL.
+     * @param string $type Specifies the menu item type
+     * @return array Returns an array
      */
+      public static function getMenuTypeInfo($type)
```
Comment (LukeTowers) [inline models/Channel.php:100]: Indentation should be four spaces, not 6

File: models/Channel.php [hunk 3]
```diff
@@ -78,8 +81,219 @@ class Channel extends Model
...
+                  $cmsPages[] = $page;
+              }
+
+              $result['cmsPages'] = $cmsPages;
+          }
+
+          return $result;
+      }
+
+    protected static function listChildrenOptions()
```
Comment (LukeTowers) [inline models/Channel.php:146]: Could you add a docblock and return type here?

File: Plugin.php (hunk 1 above)
Comment (RomainMazB) [reply Plugin.php:119]: I basically followed the blog implementation here.
https://github.com/wintercms/wn-blog-plugin/blob/main/Plugin.php#L156-L170

Comment (RomainMazB) [reply Plugin.php:118]: Ping @bennothommo

**Comment by bennothommo** [comment]
@RomainMazB let me know when this is ready and I'll review.

Would you mind also reverting the `version.yaml` changes? I prefer to modify that file in separate commits so it's easier to track in the logs and review the upgrade notes specifically.

**Pull request closed without merging**